### PR TITLE
doc: Some clarifications on signatures

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

Got asked on some questions, this is to clarify them in the doc for future reference. 
## Test Plan 

Peer review
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui/offline_signing.md [hunk 1]
```diff
@@ -18,13 +18,14 @@ Raw tx_bytes to execute: $TX_BYTES
 ```
 
 ## Step 2: Sign the data
-This can be done elsewhere in your signing device or implemented in languages of your choice. Sui accepts signatures for ECDSA Secp256k1, ECDSA Secp256r1 and pure Ed25519.
+This can be done elsewhere in your signing device or implemented in languages of your choice. Sui accepts signatures for ECDSA Secp256k1, ECDSA Secp256r1, pure Ed25519 and native multisig ([Read more](https://github.com/MystenLabs/sui/blob/d0aceaea613b33fc969f7ca2cdd84b8a35e87de3/crates/sui/multisig.md)).
```
Comment (kchalkias) [inline crates/sui/offline_signing.md:21]: I'd mention them with the order we have internally (order by flag number)

File: crates/sui/offline_signing.md [hunk 2]
```diff
@@ -18,13 +18,14 @@ Raw tx_bytes to execute: $TX_BYTES
...
 
 An accepted ECDSA Secp256k1 andd Secp256r1 signature follows:
-1. The signature must be of length 64 bytes in the form of `[r, s]` where the first 32 bytes are `r`, the second 32 bytes are `s`.
-2. The `r` value can be between 0x1 and 0xFFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFE BAAEDCE6 AF48A03B BFD25E8C D0364140 (inclusive).
-3. The `s` value must be in the lower half of the curve order, i.e. between 0x1 and 0x7FFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF 5D576E73 57A4501D DFE92F46 681B20A0 (inclusive).
-4. Ideally, the signature must be generated with deterministic nonce according to [RFC6979](https://www.rfc-editor.org/rfc/rfc6979).
+1. The signature must be committed to the SHA256 [SHA-2](https://en.wikipedia.org/wiki/SHA-2) hash of the transaction data.
```
Comment (kchalkias) [inline crates/sui/offline_signing.md:24]: We have to make it more clear somewhere if the signing api gets the actual tx bytes for all algorithms and the SHA256 is internal details to the ECDSA (to conform with the standard). We recently had a community question if when we sign with ECDSA we should pass the sha256(tx_bytes) as input. Additionally a hardware wallet provider asked what to present to the user's screen when we sign a transaction, it's blake2 or sha256 hash (when ECDSA is used) for human visual verification?

Comment (joyqvq) [reply crates/sui/offline_signing.md:24]: well it depends on the signing API that does not belong to us in offline signing this is not part of the ecdsa standard (e.g. ethereum uses keccak256). we (fastcrypto) always sign on the raw message with default hash = SHA256, but not every library. 

in fact i noticed this from pysui that ppl dont know what hash to use: https://github.com/FrankC01/pysui/blob/main/pysui/sui/sui_crypto.py#L148

Comment (joyqvq) [reply crates/sui/offline_signing.md:24]: a hardware wallet can display the human readable raw message, but they need to hash the data internally first when they pass to the signing library. it is correct that our verification runs on raw message and assumes on a default hash. https://github.com/MystenLabs/fastcrypto/blob/e6161f9279510e89bd9e9089a09edc018b30fbfe/fastcrypto/src/secp256k1/mod.rs#L121

**Review by kchalkias — APPROVED** [review]

**Pull request merged**